release-manifest-spec.md

```diff
@@ -0,0 +1,147 @@
+# Release Manifest Specification
+
+This document defines the specification for the *Release Manifest*.  The
+release manifest provides metadata about the package and TODO
+
+
+## Guiding Principles
+
+The release manifest specification makes the following assumptions about the
+document lifecycle.
+
+1. Release manifest documents will generated programatically by package management software as part of the release process.
+2. Release manifests will be consumed by package managers.
+3. Release manifests will typically **not** be stored alongside the source, but rather by package registries *or* referenced by package registries and stored in something akin to IPFS.
+
+## Format
+
+The canoonical format for the package manifest is a JSON document containing a
+single JSON object.  
+
+
+## Filename
+
+Convention is to name this document `epm.json` which is short for *'ethereum
+package manifest'*.
+
+## Document Overview
+
+The following fields are defined as data that *may* be included in a package
+manifest.  Custom fields should be prefixed with `x-` to prevent collision with
+new fields introduced in future versions of the specification.
+
+- manifest version
+- package manifest
+- version
+- sources
+- chain
+- contracts
+- build dependencies
+
+## Document Specification
+
+### Manifest Version: `manifest_version`
+
+
+The `manifest_version` field defines the version of the ethereum release manifest
+specification (this document) that this manifest conforms to. All release manifests
+**must** include this field.
+
+* Key: `manifest_version`
+* Type: Integer
+* Allowed Values: `1`
+
+
+### Version: `version`
+
+The `version` field declares the version number of this release.  This value
+**should** be conform to the [semver](http://semver.org/) version numbering
+specification.  All manifests must **include** this field.
+
+* Key: `version`
+* Type: String
+
+### Sources: `sources`
+
+The `sources` field declares a list of URIs which contain the source code for
+this release.
+
+* Key: `sources`
+* Type: List of Strings
+
+
+### Chain: `chain`
+
+The `chain` field declares the blockchain that the contract addresses apply to.
+The chain is defined by a hash with block numbers as keys and block hashes as
+values.  Both block numbers and hashes *must* be hexidecimal encoded.
+Convention is to include the genesis block under the key `0x00` and the latest
+observable block at the time of release.
+
+* Key: `chain`
+* Type: Hash(BlockNumber: BlockHash)
```

Comment:
I definitely agree with @AFDudley that we should implement something along the lines of the ChainID/URI scheme here...it is a very flexible approach and effective approach to working with multiple chains. I think the terms are being overly loaded here. I think the main point of contention is the ability to get EXACTLY what we are looking for in any chains context...which is all very doable with this. I think doing something along this line will enable solidity contracts to be ported over to areas like Rootstock (if for whatever reason they want to do that) or to permissioned chains, or to EVM forked projects. 

Alternatively as was suggested in the meeting on Wednesday, we could utilize IPFS to make this happen as its hashing could point to a specific tx in a specific block in a specific chain.